### ISSUE TITLE
Incorrectly handling _root.data

### ISSUE THREAD

**Issue by lsthornt** [kiliman/remix-express-vite-plugin#19]
I have a route configuration that looks like the following:
```
routes/_app.tsx <== has middleware for all nested routes
routes/_app._index.tsx
```

When I am on the `_app._index` page (path `/`), and I click a link that directs to the current page, a request is fired targeting `/_root.data?_routes=routes%2F_app`. The middleware added to `_app` fails to fire.

---

I am still gaining familiarity with the routing internals, but my attempt at a diagnosis is as follows:
- Any request to `/_root.data` replaces the pathname with a naked `"/"` [in the middleware request handler](https://github.com/kiliman/remix-express-vite-plugin/blob/main/packages/remix-create-express-app/src/middleware.ts#L64)
- The rewritten `"/"` pathname causes a failure to match when [matching against the request.url](https://github.com/kiliman/remix-express-vite-plugin/blob/main/packages/remix-create-express-app/src/middleware.ts#L82)

**Comment by kiliman** [comment]
Do you have `clientLoaders` or `shouldRevalidate`? These cause _Single Fetch_ to call loaders differently. I haven't implemented handling those yet.

If you could create a simple repro, I'd appreciate it.

**Comment by lsthornt** [comment]
Yes I have `shouldRevalidate = () => false;` on my root Route, as it is loading ENV variables according to [this pattern](https://remix.run/docs/en/main/guides/envvars#browser-environment-variables)

Removing that changes removes the search string from the request, so it is just requesting `/root.data`, so that part of the original report can be considered a red herring.

---

Here is a minimal stackblitz to repro: https://stackblitz.com/edit/github-spaqp1-nxqwni
- two routes, `index` and `test`, both are nested within the `_app` pathless route/layout
- middleware is added in the `_app` route
- navigating to `test` invokes the middleware, `index` does not

**Comment by lsthornt** [comment]
Digging further here, just posting findings as I learn...
- the `matches` array is correct actually. It matches `root`, `routes/_app`, and `routes/_app/_index` as expected
- the issue comes when we later [overwrite the matches array](https://github.com/kiliman/remix-express-vite-plugin/blob/main/packages/remix-create-express-app/src/middleware.ts#L86)
- `routes/_app` no longer in the matches array, so its middleware is omitted

My naïve fix would be to simply remove that conditional that forces the matches array to only the root route, but I have to assume it was added for a reason.

As is, I believe it'll cause a failure any time you use a root-level `_index` route, whether or not it is nested inside a pathless route as in my case

**Comment by kiliman** [comment]
Ugh. Thanks for looking into it. I can't say off the top of my head why I did that. It's most likely something came up when I was implementing it. Does removing that fix your use case?

I won't have time until this weekend to really get into it, so thanks for doing the grunt work.

**Comment by lsthornt** [comment]
Yeah, removing that conditional does resolve the issue in my app.

I've been trying to think about the hypothetical case that the conditional protects against—requesting `root.data` but _not_ wanting to match to `_index`—but I'm not coming up with anything. Maybe you know of some special case where `root.data` is requested outside of the normal app navigate/fetcher flow?

**Comment by kiliman** [comment]
Fixed in v0.3.8

**Comment by lsthornt** [comment]
Having a new but similar matching issue now after upgrading to `0.3.10` today.

I've updated the Stackblitz from before: https://stackblitz.com/edit/github-spaqp1-nxqwni
- Middleware attached on `_app` layout
- Middleware fires on initial SSR request
- Middleware does not fire when transitioning between leaf routes

In my app (not Stackblitz), when I log the matches array on route transition, I only get a single match (the root route).

---

[This commit](https://github.com/kiliman/remix-express-vite-plugin/commit/e0ee35cd4d017abebefff0d97ec43069b39fe881) seems to be the culprit. `req.url` is used a few lines later for matching against `routes`.

As before, I'm unsure of what prompted the change initially, but patching in a `req.url = req.originalUrl` after that line does fix the bug.
